apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerConversionOptions.cs

```diff
@@ -19,21 +19,32 @@ namespace Google.Cloud.Spanner.Data
     /// </summary>
     internal class SpannerConversionOptions
     {
-        // Predefined instances; these will change as the class grows, but hopefully
-        // for most cases we can avoid creating new instances.
-        private static readonly SpannerConversionOptions s_useDBNullForNull = new SpannerConversionOptions(true);
-        private static readonly SpannerConversionOptions s_useClrDefaultForNull = new SpannerConversionOptions(false);
-
-        internal static SpannerConversionOptions Default { get; } = s_useDBNullForNull;
+        internal static SpannerConversionOptions Default { get; } = new SpannerConversionOptions(true, false, false);
 
         /// <summary>
         /// True to return DBNull.Value for null values; false to return a null reference.
         /// </summary>
         internal bool UseDBNull { get; }
 
-        private SpannerConversionOptions(bool useDBNull)
+        /// <summary>
+        /// True to use <see cref="SpannerDbType.Numeric"/> as default SpannerDbType for decimal values; false to use <see cref="SpannerDbType.Float64"/>.
+        /// This should be used, only when working with Spanner Google Standard SQL (GSQL) dialect database.
+        /// </summary>
+        internal bool UseSpannerNumericForDecimal { get; }
+
+        /// <summary>
+        /// True to use <see cref="SpannerDbType.PgNumeric"/> as default SpannerDbType for decimal values; false to use <see cref="SpannerDbType.Float64"/>.
+        /// This should be used, only when working with Spanner PostgreSQL dialect database.
+        /// </summary>
+        internal bool UsePgNumericForDecimal { get; }
```

Comment:
Is it ever useful to specify both? Would it perhaps make things simpler to have a single parameter of "DefaultTypeForDecimal=PgNumeric" or "DefaultTypeForDecimal=Numeric"?